app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Save.php

```diff
@@ -17,22 +17,46 @@ class Save extends \Magento\Catalog\Controller\Adminhtml\Product\Set
      * @var \Magento\Framework\Controller\Result\JsonFactory
      */
     protected $resultJsonFactory;
-
+    
+    /*
+     * @var \Magento\Eav\Model\Entity\Attribute\SetFactory
+     */
+    private $attributeSetFactory;
+    
+    /*
+     * @var \Magento\Framework\Filter\FilterManagerFactory
+     */
+    private $filterManagerFactory;
+    
+    /*
+     * @var \Magento\Framework\Json\Helper\DataFactory
+     */
+    private $jsonHelperFactory;
+    
     /**
      * @param \Magento\Backend\App\Action\Context $context
      * @param \Magento\Framework\Registry $coreRegistry
      * @param \Magento\Framework\View\LayoutFactory $layoutFactory
      * @param \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory
+     * @param \Magento\Eav\Model\Entity\Attribute\SetFactory $attributeSetFactory
+     * @param \Magento\Framework\Filter\FilterManagerFactory $filterManagerFactory
+     * @param \Magento\Framework\Json\Helper\DataFactory $jsonHelperFactory
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
         \Magento\Framework\Registry $coreRegistry,
         \Magento\Framework\View\LayoutFactory $layoutFactory,
-        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory
+        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
+        \Magento\Eav\Model\Entity\Attribute\SetFactory $attributeSetFactory = null,
+        \Magento\Framework\Filter\FilterManagerFactory $filterManagerFactory = null,
+        \Magento\Framework\Json\Helper\DataFactory $jsonHelperFactory = null
```

Comment:
We don't have any places where this class was created via factory.

Example: https://github.com/magento/magento2/blob/9c14af7a97bcf338662de365d2613c7ca9dbf8c6/app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php#L44